extensions/Maps/includes/Maps_Geocoder.php

```diff
@@ -111,7 +111,21 @@ public function hasAlias( $alias ) {
 	 * @return array or false
 	 */
 	public function geocode( $address ) {
-		$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+
+		// MAIN-6654: cache requests to this API because it has daily limits
+		global $wgMemc;
+		$key = wfMemcKey( "geocoder", $address );
+		$response = $wgMemc->get( $key );
+		if ( empty($response) ) {
+			$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+			// MAIN-6654: make sure we do not return or store a bad response because it causes errors later
+			if ( substr( $response, "OVER_QUERY_LIMIT" ) !== false ) {
```

Comment:
After double  checking, I don't think this solution help us.
Caching is already here: https://github.com/Wikia/app/blob/24375329bce0086d595ea9a3d6851993dd1abcea/extensions/Maps/includes/Maps_Geocoders.php#L197-L197.